content/training/R_Basics/Advanced_Functions.Rmd

```diff
@@ -196,10 +196,150 @@ This is often used to inspect which rows are unique to a data frame. An anti joi
+                        y = "Cases (n)", 
+                        title = "Cases by week of onset",
+                        subtitle = sprintf("Source: MSF data from %s", reporting_week)
+                       ) 
+```
+
+ 
+
+
+## For loop  
+
+A **for loop** is used to repeat or iterate code, making repetitive tasks more efficient. The "loop" repeats a series of commands, altering them slightly to account for different input values that are stored in a vector. This vector, or sequence of values, can be the values within a data frame variable. 
+
+In the outbreak templates, a **for loop** is used at the end of the **Place** analyses to create bar graphs and maps of regional attack rates *for each epidemiological week*.  
+
+The basic structure of a **for loop** is as follows:  
+
+* **Setup**, including :  
+     * The keyword `for` (note the lowercase)  
+     * Parentheses containing:  
+          * A placeholder term (e.g. "i", "value", or another relevant short word)  
+          * The keyword `in`  
+          * The name of the vector to loop through  
+     * Opening bracket `{`  
+
+* **Actions** and commands to be performed for each item in the vector  
+     * If the item is referenced in any command, use the placeholder term instead.  
+     
+* **Closing** - A closing bracket `}`
+      
+
+A basic example **for loop** is below. `number_list` is defined as a vector of four numbers. The **for loop** is set-up to examine each item ("i") in `number_list`. The only command in the **for loop** is to print the value of "i" plus 2. Finally, the brackets are closed. As the loop executes, it produces one output for each item in `number_list`.
+
+```{r}
+number_list <- c(102, 53, 14, 88)
+
+for (i in number_list) {
+     print(i + 2)
+}
+```
+
+In the outbreak templates, the **for loop** is more complex than the simple example above, but follows the same basic structure.  
+
+The loop is set-up for each item "i" of all the unique values in the variable `epiweek` within the data frame `cases`. This vector is a list of the epidemiological weeks with no repeated values. If you want to see this vector of epidemiological weeks, highlight and run just `unique(cases$epiweek)`.  
+
+Numerous commands are located within the loop's brackets `{ }`, but the most important is the first: `this_ar <- filter(ar, epiweek == i)`. This command restricts the large data frame `ar` to only the epiweek currently under consideration by the **for loop**. *Thus, the data frame `this_ar` will be different for each iteration of the **for loop** *. All the subsequent steps (producing a map and a barplot) are performed on this filtered dataset.  
+
+The final step produces a `cowplot()` visualization that includes the map and the barplot for the current epiweek "i". After producing the cowplot, the loop advances "i" to the next epiweek item in the vector, repeats all the commands in the brackets `{ }`, and produces a new cowplot.
```

Comment:
Suggestion to change the wording here as cowplot is not necessarily a specific plot type, it's just a way of combining plots on a grid. 

```suggestion
The final step creates a combined plot of the map and barplot for the current epiweek "i" using the `plot_grid()` function from the **cowplot** package. This plot is aligned horizontally based on the top and bottom of each plot. After producing the combined plot, the loop advances "i" to the next epiweek item in the vector, repeats all the commands in the brackets `{ }`, and produces a new combined plot.
```